### ISSUE TITLE
Problem: Using Arduino DUE and SSD1306 with HW I2c not working

### ISSUE THREAD

**Issue by TKaluza** [olikraus/u8g2#304]
Hi there! 
Wonderfull library, I love it and thank you for making it!

I have used my Arduino MEGA with your library and everything worked fine. 
Then switched for more computing power to the Arduino Due and I do net get the display working. 

The Arduino DUE ist connected with the Programming Port. Using Pins 20 (SDA) and 21 (SCL) as the I2C connection ports. Those are connected to the display with a level shifter which has at LV 3.3V and at HV 5.0V. The Display itself is connected with 5V from the 5V port. 

After every upload of the sketch I hit the reset "Button". Cables not broken. Display works on MEGA.

With this setup and using the [I2C-Scanner](https://playground.arduino.cc/Main/I2cScanner) I get:
```
Scanning...
I2C device found at address 0x3C  !
done
```

So I think the setup is valid? (It got power and the DUE seems to find it - nothing else is connected)

As a constructor I am using the same as with the MEGA: 

> `U8G2_SSD1306_128X64_NONAME_F_2ND_HW_I2C mydisplay(U8G2_R0,/* reset=*/ U8X8_PIN_NONE); ` 

My bad I tried the second I2C, but I meant:
`U8G2_SSD1306_128X64_NONAME_F_HW_I2C mydisplay(U8G2_R0,/* reset=*/ U8X8_PIN_NONE);`

The Display stays black. I also tried to power it from different power sources all 5V or connecting the I2C pins directly. 

I saw, that you are using the DUE yourself, that's why I am asking here. 
If there's anything unclear -- I'm happy to provide further information. 
Thank you very much!

**Comment by olikraus** [comment]
Pins 20 (SDA) and 21 (SCL) are the first I2C interface, istn't it?
Maybe try
`U8G2_SSD1306_128X64_NONAME_F_HW_I2C mydisplay(U8G2_R0,/* reset=*/ U8X8_PIN_NONE);
`

Just to ensure, that the display it self works, you can also try the more slower SW I2C constructor:
`U8G2_SSD1306_128X64_NONAME_F_SW_I2C mydisplay(U8G2_R0, 21, 20,/* reset=*/ U8X8_PIN_NONE);
`

**Comment by TKaluza** [comment]
Thanks for the fast replay. 

I'm sorry I meant the first I2C. 

I also tried your **SW** you've suggested, but It does not work. 

What I'm curious about is, if I have fried the Pins, it shouldn't be able to detect a I2C device, should it?

It is my frist time with the DUE, so using the Programming Code for powering the System is okay, or do I need to power it differently? 
I also tried the sparkfun library and example for the SSD1306 but it also doesn't work. So I suppose your code is probably not the problem but something with my DUE. 
If you have any thought about that, I would be delighted to here them :). 

Thanks

**Comment by olikraus** [comment]
Strange, yes if the scanner finds the OLED, then the interface should work. Did you try using the OLED directly, without levelshifter?  Also use 3.3 power when trying this

**Comment by TKaluza** [comment]
Now come a  few pictures - sorry. 
![img_20170711_220307](https://user-images.githubusercontent.com/17787801/28089053-fd1b617c-6687-11e7-8435-143a479ff107.jpg)
**Pic 1**: 
Working with connection to _MEGA_

![img_20170711_220313](https://user-images.githubusercontent.com/17787801/28089054-fd1b7fc2-6687-11e7-8378-2f4c5be6076d.jpg)
**Pic 2**: 
Just better overview of the cables

![img_20170711_220519](https://user-images.githubusercontent.com/17787801/28089056-fd1f930a-6687-11e7-9168-36941311e852.jpg)
**Pic 3**: 
Now connceted to the _DUE_ trying both `HW` or `SW`

![img_20170711_220650](https://user-images.githubusercontent.com/17787801/28089058-fd26da48-6687-11e7-8a65-114ef015a4d6.jpg)
**Pic 4**: 
Checking level shifter - seems to output something. 

![img_20170711_220923](https://user-images.githubusercontent.com/17787801/28089057-fd209f52-6687-11e7-8e98-60f1b33aa618.jpg)
**Pic 5**:
Checking now on the _DUE_  with 3.3V and direct connection. The SDA and SCL are showing some voltage. 

![img_20170711_221255](https://user-images.githubusercontent.com/17787801/28089059-fd368a2e-6687-11e7-9f02-a7899d28c23d.jpg)
**Pic 6**: 
Connecting the VCC just to 3.3V but the SCL and SDA to the _MEGA_. It works.

![img_20170711_221619](https://user-images.githubusercontent.com/17787801/28089060-fd39ac5e-6687-11e7-9382-805e1ca21862.jpg)
**Pic 7**: 
And the I2C Scanner running on the _MEGA_ finds it. 

![img_20170711_221647](https://user-images.githubusercontent.com/17787801/28089061-fd3b162a-6687-11e7-958c-3848f540fd01.jpg)
**Pic 8**: 
Setting up the _DUE_ with the I2C Scanner

![img_20170711_221828](https://user-images.githubusercontent.com/17787801/28089062-fd440f6e-6687-11e7-88fa-5d8794a51c85.jpg)
**Pic 9**: 
It can also find the display. 

---

Do you have any idea or thought what I'm doing wrong? 
If not, absolutly no problem. I don't have a clue. Trying some other code tomorrow. But I already testet the adafruit lib with the example. I'll also trying other display-modul tomorrow. 

Thank you very much.

**Comment by olikraus** [comment]
So you tried all three constructors with Due (1st HW, 2nd HW and SW I2C). Each of them you tried with a proper I2C level shifter, correct?
With each kind of wiring, the OLED is detected by the I2C scanner, but none of the OLED libs works.

I once had a similar case, where an OLED works with one board, but did not work with another (altough I do not remember, whether Due was involved here).

Maybe you should try to use an external power supplier, but all in all, I do not know what else could be the reason.

**Comment by YerBabiE** [comment]
I've just encountered the same issue, I was using one brand of display fine with a wemos D1 mini using the 5V rail etc. then I switched to this new screen, and it will not light up it says www.diymore.cc on the back. If I scan for 12c interfaces it finds it, but it just will not work when given commands. I have tried on a few screens and few D1's all they are do the same thing... very frustrating. Did you end up getting a solution to this?
